apps/opentelemetry_experimental/src/otel_log_handler.erl

```diff
@@ -28,98 +28,143 @@
          adding_handler/1,
          removing_handler/1,
          changing_config/3,
-         filter_config/1,
          report_cb/1]).
 
 -export([init/1,
          callback_mode/0,
          idle/3,
          exporting/3,
-         handle_event/3]).
+         terminate/3]).
 
 -type config() :: #{id => logger:handler_id(),
-                    regname := atom(),
-                    config => term(),
+                    reg_name := atom(),
+                    config => otel_log_handler_config(),
                     level => logger:level() | all | none,
                     module => module(),
                     filter_default => log | stop,
                     filters => [{logger:filter_id(), logger:filter()}],
                     formatter => {module(), logger:formatter_config()}}.
 
--define(DEFAULT_CALL_TIMEOUT, 5000).
--define(DEFAULT_MAX_QUEUE_SIZE, 2048).
--define(DEFAULT_SCHEDULED_DELAY_MS, timer:seconds(5)).
--define(DEFAULT_EXPORTER_TIMEOUT_MS, timer:minutes(5)).
+-type otel_log_handler_config() ::
+        #{max_queue_size => max_queue_size(),
+          max_export_batch_size => max_export_batch_size(),
+          exporting_timeout_ms => exporting_timeout_ms(),
+          scheduled_delay_ms => scheduled_delay_ms(),
+          check_table_size_ms => check_table_size_ms(),
+          exporter => exporter_config()}.
+
+-type max_queue_size() :: non_neg_integer() | infinity.
+-type exporter_config() :: module() | {module(), Config :: term()} | undefined.
+-type max_export_batch_size() :: non_neg_integer().
+-type exporting_timeout_ms() :: non_neg_integer().
+-type scheduled_delay_ms() :: non_neg_integer().
+-type check_table_size_ms() :: non_neg_integer().
+
+-define(SUP, opentelemetry_experimental_sup).
 
 -define(name_to_reg_name(Module, Id),
         list_to_atom(lists:concat([Module, "_", Id]))).
+-define(CURRENT_TABLES_KEY(Name), {?MODULE, current_table, Name}).
+-define(TABLE_NAME(TN), lists:concat([TN, "_", erlang:pid_to_list(self())])).
```

Comment:
this is a risk of atom leak if the process restarts a lot of times.

maybe just use `otel_log_handler_table_1` and `otel_log_handler_table_2`

Reply:
Thanks, this indeed looks unnecessary, this was copied from batch processor. Will fix it, <module>_<handler_id>_table_<N> must be enough..

Reply:
Fixed